Add `Claim all` button with saga

ArmandoGraterol opened the pull request:
## Description

- Added "Claim all" button to the stakes tab, now you should be able to claim all the stakes there. I tried to simplify the related saga as much as I could, let me know if you come up with something that can be improved.

**New stuff** ✨

* Added `ClaimAllButton` component and connected it to the stakes tab

**Changes** 🏗

* Updated `transform` on `FinalizeMotionAndClaimWidget`
* Updated `claimMotionRewards` to accept multiple motion ids

![FireShot Capture 533 - Colony - localhost](https://user-images.githubusercontent.com/18473896/153040279-87b75db9-c17f-41c2-872d-6c5c6bed7b11.png)


Resolves #3155

arrenv commented:
Hey @ArmandoGraterol, just confirming, this is only for the 'Claim all' button functionality. Not the copy next to the button or anything else?

ArmandoGraterol commented:
> Hey @ArmandoGraterol, just confirming, this is only for the 'Claim all' button functionality. Not the copy next to the button or anything else?

Uh I don't think so? Isn't that part of another issue?

arrenv commented:
@ArmandoGraterol I agree, it is not at all clear in the raised issue. I have updated it, it should include the "Your stakes" copy.

ArmandoGraterol commented:
> @ArmandoGraterol I agree, it is not at all clear in the raised issue. I have updated it, it should include the "Your stakes" copy.
> 
> I hope that is okay, otherwise, I will create a separate issue for it.

Done! Thank you for the clarification

arrenv reviewed (changes requested):
@ArmandoGraterol really nice work. I like how you used the same saga.

I tested and everything seemed to work great. However, I tested it again so that I could do a screen record and the result was the following gif. I will restart my dev environment and test again though.

![claimable-transactions-not-working](https://user-images.githubusercontent.com/33682027/153235005-59319013-bed6-4503-a66c-32d36f829f8b.gif)

ArmandoGraterol commented on src/modules/users/components/TokenActivation/TokenActivationContent/TokenActivationContent.css:
```diff
@@ -282,3 +282,35 @@
+
+.claimAllButtonSection button {
+  width: 100px;
+  font-size: --size-small;
+}
+
+.claimAllButtonSection h4 {
+  font-size: --size-smallish;
```
Oh damn you are right! Thank you

ArmandoGraterol commented:
> I tested and everything seemed to work great. However, I tested it again so that I could do a screen record and the result was the following gif. I will restart my dev environment and test again though.

Please do so! I haven't been able to reproduce that

arrenv commented:
@ArmandoGraterol I have been testing this some more and the issue is with the `useClaimableStakedMotionsQuery` query.
Curerntly, you have to reload the app for the query to update, and it also includes motions that have been staked and not finalized.

I will create a separate issue for updating the resolver.

With this PR, do you think there needs to be feedback or fallback for if the transaction fails or can't be performed. For example, in the Network tab in console, when my gif above happens, I get this error.

`message: "VM Exception while processing transaction: revert voting-rep-motion-not-claimable"`

arrenv commented:
@ArmandoGraterol I have updated the resolver, which should help with this issue - https://github.com/JoinColony/colonyDapp/pull/3175

ArmandoGraterol commented:
> @ArmandoGraterol I have updated the resolver, which should help with this issue - #3175

Nice! I also added an extra, just in case, error that was missing from the saga, and added a query to update the claim list after claiming something with the saga. So, the case in your gif should now not happen

arrenv reviewed (commented):
@ArmandoGraterol It is working great, nice!

With the changes I made to `useClaimableStakedMotionsQuery`  in https://github.com/JoinColony/colonyDapp/pull/3175 returning motions as opposed to just an array of motionIds, should we manage that in this PR or do another PR for it?

ArmandoGraterol commented:
> With the changes I made to `useClaimableStakedMotionsQuery` in #3175 returning motions as opposed to just an array of motionIds, should we manage that in this PR or do another PR for it?

I think it's more convenient if you merge yours first and then I update this one with the changes you made on that PR :)

chinins reviewed (approved):
Very nice work. 
I noticed that after claiming all I still have value in the staked pat of the token activation popover which resets to zero after I refresh. I think it would be good to refresh the query that provides these numbers (I think it's user lock query)  in the claimMotionRewards saga after the action success.
<img width="445" alt="Screenshot 2022-02-14 at 23 55 38" src="https://user-images.githubusercontent.com/34057551/153945056-a55afaa3-3846-4cbc-b988-22075602a5c0.png">

ArmandoGraterol commented:
> I noticed that after claiming all I still have value in the staked pat of the token activation popover which resets to zero after I refresh. I think it would be good to refresh the query that provides these numbers (I think it's user lock query) in the claimMotionRewards saga after the action success.

Oh yeah, I agree with you! I Thank you for bringing that up

danbr commented on src/modules/users/components/TokenActivation/TokenActivationContent/StakesTab.tsx:
```diff
@@ -4,10 +4,17 @@ import { FormattedMessage, defineMessages } from 'react-intl';
 
 const MSG = defineMessages({
+  tabTitle: {
+    id: 'users.TokenActivation.TokenActivationContent.ClaimsTab.noClaims',
```
.ClaimsTab..... - but was renamed to StakesTab. I can update when i make my changes for the dialog.

danbr reviewed (approved):
nice work - was good for me to review, as i'll be rebasing off your PR and incorporating your changes with my PR.
